common/corerequestcontext.go

```diff
@@ -88,17 +88,24 @@ func UpdateResponseStatus(ctx context.Context, status int) error {
 	return nil
 }
 
-func CoreRequestContextMiddleware(logger ...*logrus.Logger) func(next http.Handler) http.Handler {
+func CoreRequestContextMiddleware(logger *logrus.Logger) func(next http.Handler) http.Handler {
+	ctx := LoggerToContext(context.Background(), logger, nil)
+	return CoreRequestContextMiddlewareWithContext(ctx)
+}
+
+func CoreRequestContextMiddlewareWithContext(ctx context.Context) func(next http.Handler) http.Handler {
 	var ctxlogger *logrus.Logger
-	if len(logger) == 1 {
-		ctxlogger = logger[0]
+	if getCoreContext(ctx) != nil {
```

Comment:
I'd modify `common.GetLoggerFromContext` to return `nil` if no logger is stored, then just call this method directly from here.